src/pages/DetailPage.js

```diff
@@ -11,10 +11,10 @@ const DetailPage = () => {
   const { state } = useContext(GlobalContext);
   const { id } = useParams();
   const { categories } = state;
-
+  console.log({categories})
   let navigate = useNavigate();
 
-  const selectedCategory = categories.find((item) => item.id == id);
+  const selectedCategory = categories[id];
```

Comment:
much easier to find the `selectedCategory` now that we're using an object instead of an array - its now a simple object lookup with `categories[id]`